docs/manual/linux-udev.md

```diff
@@ -3,19 +3,93 @@ layout: default
 title: Configuring udev on Linux
 parent: User manual
 nav_order: 6
+has_toc: true
 redirect_from:
   - /getting-started/linux-udev.html
 ---
 
-# Configuring udev on Linux
+# Configuring `udev` rules for VIA and Vial on Linux
 
-In order for your keyboard to be detected on Linux, you need to set up a custom udev rule. Open a terminal and execute the following commands to set this up:
+In order for your keyboard to be detected by both VIA and Vial on Linux, you need to set up a custom `udev` rule. Both use the `hidraw` linux driver to directly communicate configurations to your keyboard. In most situations it should be safe to use a generalized `udev` rule for all `hidraw` devices, but in other cases where you prefer or are required to restrict access on a device basis, it is possible to create a `udev` rule specifically for your keyboard.
+
+The following guides will show you how to implement these `udev` rules. You will require root access to create the files necessary.
+
+## Vial `udev` rule
+
+For a generalized access rule for any device with Vial firmware, write this text to `/etc/udev/rules.d/99-vial.rules`:
+
+```
+KERNEL=="hidraw*", SUBSYSTEM=="hidraw", ATTRS{serial}=="vial:f64c2b3c", MODE="0660", GROUP="users", TAG+="uaccess", TAG+="udev-acl"
```

Comment:
Is there a way to make this match on a substring and not exact match? since vial magic can be anywhere inside the serial number

Reply:
What exactly do you mean by this? I'm guessing that the `vial:f64c2b3c` string gets appended or placed within an existing serial number? If you could describe how the magic string actually gets inserted it would be helpful.

Reply:
Ah, okay. This is in the `udev(7)` man page:

       Most of the fields support shell glob pattern matching and alternate
       patterns. The following special characters are supported:

       *
           Matches zero or more characters.

       ?
           Matches any single character.

       []
           Matches any single character specified within the brackets. For
           example, the pattern string tty[SR] would match either ttyS or
           ttyR. Ranges are also supported via the - character. For example,
           to match on the range of all digits, the pattern [0-9] could be
           used. If the first character following the [ is a !, any characters
           not enclosed are matched.

       |
           Separates alternative patterns. For example, the pattern string
           abc|x* would match either abc or x*.

I believe it should be as simple as surrounding the magic string with star globs.

Reply:
added star globs to the magic string.